snapcraft/plugins/rust.py

```diff
@@ -223,13 +235,27 @@ def _build_env(self):
 
         return env
 
-    def _get_toolchain(self) -> str:
+    def _get_toolchain(self) -> Optional[str]:
+        """Return the toolchain to use.
+
+        If a rust-toolchain file is present, None will be returned.
+        If a rust-toolchain file is not present and neither rust-version
+        nor rust-channel are set, then stable will be returned.
+        """
+        if self.options.source_subdir:
+            sourcedir = os.path.join(self.sourcedir, self.options.source_subdir)
+        else:
+            sourcedir = self.sourcedir
+        rust_toolchain_path = os.path.join(sourcedir, "rust-toolchain")
+
         if self.options.rust_revision:
```

Comment:
I wonder if it should error if both revision and channel are specified in the config?  Otherwise it should probably spell out the precedence in the documentation.

Reply:
I will do a follow-up for this, it is documented though